pkg/controller/deliverable/reconciler.go

```diff
@@ -154,41 +156,46 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		}
 	} else {
 		if log.V(logger.DEBUG).Enabled() {
-			for _, stampedObject := range stampedObjects {
+			for _, resource := range realizedResources {
 				log.V(logger.DEBUG).Info("realized object",
-					"object", stampedObject)
+					"object", resource.StampedRef)
 			}
 		}
 		r.conditionManager.AddPositive(ResourcesSubmittedCondition())
 	}
 
-	r.trackDependencies(deliverable, selectedTemplates, serviceAccountName, serviceAccountNS)
+	r.trackDependencies(deliverable, realizedResources, serviceAccountName, serviceAccountNS)
 
 	var trackingError error
-	if len(stampedObjects) > 0 {
-		for _, stampedObject := range stampedObjects {
-			trackingError = r.StampedTracker.Watch(log, stampedObject, &handler.EnqueueRequestForOwner{OwnerType: &v1alpha1.Deliverable{}})
-			if trackingError != nil {
-				log.Error(err, "failed to add informer for object",
-					"object", stampedObject)
-				err = controller.NewUnhandledError(trackingError)
-			} else {
-				log.V(logger.DEBUG).Info("added informer for object",
-					"object", stampedObject)
-			}
+	for _, resource := range realizedResources {
+		if resource.StampedRef == nil {
+			continue
+		}
+		obj := &unstructured.Unstructured{}
+		obj.SetGroupVersionKind(resource.StampedRef.GroupVersionKind())
+
+		trackingError = r.StampedTracker.Watch(log, obj, &handler.EnqueueRequestForOwner{OwnerType: &v1alpha1.Deliverable{}})
+		if trackingError != nil {
+			log.Error(err, "failed to add informer for object",
+				"object", resource.StampedRef)
+			err = controller.NewUnhandledError(trackingError)
+		} else {
+			log.V(logger.DEBUG).Info("added informer for object",
+				"object", resource.StampedRef)
 		}
 	}
 
-	return r.completeReconciliation(ctx, deliverable, err)
+	return r.completeReconciliation(ctx, deliverable, realizedResources, err)
 }
 
-func (r *Reconciler) completeReconciliation(ctx context.Context, deliverable *v1alpha1.Deliverable, err error) (ctrl.Result, error) {
+func (r *Reconciler) completeReconciliation(ctx context.Context, deliverable *v1alpha1.Deliverable, realizedResources []v1alpha1.RealizedResource, err error) (ctrl.Result, error) {
 	log := logr.FromContextOrDiscard(ctx)
 	var changed bool
 	deliverable.Status.Conditions, changed = r.conditionManager.Finalize()
 
 	var updateErr error
-	if changed || (deliverable.Status.ObservedGeneration != deliverable.Generation) {
+	if changed || (deliverable.Status.ObservedGeneration != deliverable.Generation) || !reflect.DeepEqual(deliverable.Status.Resources, realizedResources) {
```

Comment:
nit: I usually see folks using [`equality.Semantic.DeepEqual`](https://github.com/kubernetes/apimachinery/blob/2936d3f03931b9c06a641799605d8e806cb2a58b/pkg/api/equality/semantic.go), but, tbh, I'm not sure how significant that is for what we're achieving here

Reply:
I'm not sure this is the correct DeepEqual to use? We are comparing two []v1alpha1.RealizedResource. That seems to be intended for full api objects? We use reflect.DeepEqual else where in the code to compare structs/[]structs.

Reply:
I see, yeah makes sense 👍 👍  thank you!